mono/metadata/icall.c

```diff
@@ -7330,35 +7326,29 @@ ves_icall_System_IO_get_temp_path (MonoError *error)
 
 #ifndef PLATFORM_NO_DRIVEINFO
 MonoBoolean
-ves_icall_System_IO_DriveInfo_GetDiskFreeSpace (MonoString *path_name, guint64 *free_bytes_avail,
+ves_icall_System_IO_DriveInfo_GetDiskFreeSpace (const gunichar2 *path_name, gint32 path_name_length, guint64 *free_bytes_avail,
 						guint64 *total_number_of_bytes, guint64 *total_number_of_free_bytes,
 						gint32 *error)
 {
-	gboolean result;
-
 	*error = ERROR_SUCCESS;
 
-	result = mono_w32file_get_disk_free_space (mono_string_chars_internal (path_name), free_bytes_avail, total_number_of_bytes, total_number_of_free_bytes);
+	if (free_bytes_avail)
+		*free_bytes_avail = (guint64)-1;
+
+	if (total_number_of_bytes)
```

Comment:
There is no need to check for `NULL`.

Reply:
Generally removed new/existing null checks and added g_assert.